src/factory/ChartFactory.js

```diff
@@ -0,0 +1,397 @@
+import React, { useMemo } from 'react';
+
+import BarChart from '../BarChart';
+import PieChart from '../PieChart';
+import NestedProportionalAreaChart from '../NestedProportionalAreaChart';
+import NumberVisuals from '../NumberVisuals';
+
+import aggregateData from './utils/aggregateData';
+import propTypes from '../propTypes';
+import withVictoryTheme from '../styles/withVictoryTheme';
+
+function ChartFactory({
+  theme,
+  visual: {
+    type: visualType,
+    label,
+    reference: { label: referenceLabel } = {},
+    aggregate,
+    unit = '',
+    subtitle,
+    description,
+    props: visualProps = {},
+    statistic = {}
+  },
+  data,
+  refrence: { label: propRefrenceLabel },
+  isComparison,
+  comparisonData,
+  refrenceData,
+  profiles
+}) {
+  const key =
```

Comment:
1. They way we use the factories, each chart already has an `id` in the JSON data, we should use that as `key`.
2. If non supplied, lets just use `shortid` to generate one.